### ISSUE TITLE
training yolo v8 on a custom dataset, but retaining the ability to detect the person class

### ISSUE THREAD

**Issue by luise15** [ultralytics/ultralytics#7264]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello, I'm working on detecting people in security cameras located at a height of 12 meters. I'm using the YOLOv8x model. There are many people that are not detected by the YOLOv8x model with its base weights. Therefore, I need to fine-tune YOLOv8 on my dataset. I am extracting frames and labeling them.

I'm facing the following situation: when I label a dataset of frames and train YOLOv8 on this custom dataset, what happens is that YOLOv8 learns to detect the labeled people in the custom dataset but loses the ability to detect people who are not labeled, which it could detect before the training.

When I create a custom dataset, I am assigning new IDs to the classes. So, I suspect that when I use the 'train' method of YOLOv8, the weights of the last layers are discarded. Can someone please explain what happens when I use the 'train' method?

On the other hand, can you please advise me on how I can retrain YOLOv8 to preserve the original class IDs, and have the model only improve its detection capability for classes it could already detect, especially the 'person' class?"

**Comment by glenn-jocher** [comment]
@luise15 hello! When you train YOLOv8 on a custom dataset, the model will focus on the classes present in your dataset. If the 'person' class is not included or is assigned a new ID, the model may not retain its original ability to detect people as it was trained on the COCO dataset.

To address this, ensure that your custom dataset includes the 'person' class with the same class ID as in the COCO dataset. During training, YOLOv8 will then fine-tune its ability to detect the 'person' class based on your new data while retaining its pre-trained knowledge.

When using the 'train' method, the model updates its weights based on the classes it sees during training. If you want to retain the detection capabilities for the 'person' class, you should include it in your custom dataset with ample examples, particularly those that represent the new perspectives and scales from your security cameras.

For more guidance on training with custom datasets and maintaining class IDs, please refer to our documentation on the Train mode. 📘🚀

**Comment by AbinasChopdar** [comment]
@luise15 has your problem been solved? 
I am also working on a similar problem where I need to fine tune the pretrained COCO weights to work on the new dataset and my focus is on a few classes. The constraint is after training, the new weight should work for both COCO and new dataset.

@glenn-jocher First of all is it possible? What should be the training parameters for better training? I trained the model for at max 10 epoch and reduced the learning rate to 0.0001. I have also tried partial training of the model. The number of classes in the pretrained yolov8 coco model is 80, I modified the model layers to detect only 10 classes. Now I want to fine tune this new modified model so that it works on Coco as well as the new dataset.

**Comment by hopeSerendipity** [comment]
@AbinasChopdar I am also faced with the same problem,I have a custom dataset and i want the model trained by the dataset can also keep the ability of detecting person.Do you have any idea about that and does the method method mentioned by @glenn-jocher work?

**Comment by AbinasChopdar** [comment]
@hopeSerendipity Fine tuning on my_dataset did not work so I trained the model with combined dataset (COCO + my_dataset) so this way it worked.

**Comment by hopeSerendipity** [comment]
@AbinasChopdar the pictures  in coco dataset are tremendous,will it affect the precison of other classes? Or you just choose part of the pictures labeled with the class you want instead of all the pictures which are labeled with the class you want.
